gcc/rust/checks/errors/borrowck/polonius/rust-polonius-ffi.h

```diff
@@ -102,11 +102,77 @@ struct FactsView
   Slice<Pair<Origin, Loan>> placeholder;
 };
 
+// Wrapper around std::vector to pass data from Rust to C++
+template <typename T> struct FFIVector
+{
+  std::vector<T> data;
+
+public:
+  void push (T new_element) { data.push_back (new_element); };
+
+  // allocates memory to a new instance and returns the pointer
+  static FFIVector *make_new () { return new FFIVector{}; }
+
+  // returns current size
+  size_t size () const { return data.size (); }
+
+  T at (size_t index) const
+  {
+    rust_assert (index < data.size ());
+    return data.at (index);
+  }
+};
+
+// Some useful type aliases
+using FFIVectorSizet = FFIVector<size_t>;
```

Comment:
nit:I would remove that alias as I think it does not make reading easier. I prefer seeing `FFIVector<size_t>` .`FFIVectorPair` and `FFIVectorTriple` are good aliases I think though :)